<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="EP" />

<meta name="date" content="2025-10-04" />

<title>Fibroblast GSEA Nature validation</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiac Sarcoidosis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Fibroblast GSEA Nature validation</h1>
<h4 class="author">EP</h4>
<h4 class="date">2025-10-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-05
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiac Sarcoidosis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20251004code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20251004)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20251004code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20251004)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Documents/Github/Cardiac
Sarcoidosis/data/Merged/Sarcoidosis_vs_DonorHeart_v2.rds
</td>
<td style="text-align:left;">
data/Merged/Sarcoidosis_vs_DonorHeart_v2.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong6ae5a3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 6ae5a3a
</a>
</p>
</div>
<div id="strongRepositoryversionstrong6ae5a3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 6ae5a3a.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    R/
    Ignored:    analysis/.DS_Store
    Ignored:    data/Merged/

Untracked files:
    Untracked:  .gitignore.gitignore
    Untracked:  Fibroblast-GSEA-Nature-Validation_files/
    Untracked:  analysis/Cellchat_Sarcoidosis_v3.Rmd
    Untracked:  analysis/Fibroblast_GSEA_Nature_Validation.Rmd
    Untracked:  analysis/Sarcoidosis_ClusterAnnotations.Rmd
    Untracked:  analysis/Sarcoidosis_GREM2.Rmd
    Untracked:  analysis/Sarcoidosis_Subanalysis.Rmd
    Untracked:  analysis/Sarcoidosis_allClusters_Validation_Nature.Rmd
    Untracked:  analysis/old files/
    Untracked:  code/gitignore

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre><code>[1] &quot;data/Merged/Sarcoidosis_vs_DonorHeart_v2.rds&quot;</code></pre>
<pre><code>
      
49601 </code></pre>
<div id="umap" class="section level2">
<h2>UMAP</h2>
<pre class="r"><code>colclusterName &lt;- c(&quot;#f4a582&quot;, &quot;#FEE08B&quot;, &quot;#3288BD&quot;, &quot;#253494&quot;,  &quot;#01665e&quot;, &quot;#66C2A5&quot;, &quot;#355C7D&quot;, &quot;#a6bddb&quot;, &quot;#fde0dd&quot;, &quot;#dd1c77&quot;)

names(colclusterName) &lt;- c(&quot;CM&quot;, &quot;Fb1&quot;,&quot;PeriFb&quot;, &quot;VSMC&quot;, &quot;BEC1&quot;,&quot;BEC2&quot;, &quot;NC1&quot;, &quot;TRM&quot;, &quot;inflMph&quot;,&quot;Lymphocytes&quot;)

seuratM$clusterName &lt;- factor(seuratM$clusterName, levels=c(&quot;CM&quot;, &quot;Fb1&quot;, &quot;PeriFb&quot;, &quot;VSMC&quot;, &quot;BEC1&quot;,&quot;BEC2&quot;, &quot;NC1&quot;, &quot;TRM&quot;, &quot;inflMph&quot;,&quot;Lymphocytes&quot;))

Idents(seuratM) &lt;- seuratM$clusterName
DimPlot(seuratM, reduction = &quot;umap&quot;,label = TRUE, cols = colclusterName, repel = TRUE, pt.size = 0.01)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>seuratFb &lt;- subset (seuratM, clusterName %in% c(&quot;Fb1&quot;, &quot;PeriFb&quot;, &quot;VSMC&quot;))

table(seuratFb$orig.ident)</code></pre>
<pre><code>
      
24740 </code></pre>
<pre class="r"><code>table(seuratFb$pat_sub)</code></pre>
<pre><code>
 DonorHeart Sarcoidosis 
      19792        4948 </code></pre>
</div>
<div id="reembedding" class="section level2">
<h2>Reembedding</h2>
</div>
<div id="fibroblast-annotation" class="section level2">
<h2>Fibroblast annotation</h2>
<pre class="r"><code>Idents (seuratFb) &lt;- seuratFb$RNA_snn_res.0.3

DimPlot(seuratFb, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>seuratFb$clusterName &lt;- &quot;clusterName&quot;
Idents(seuratFb) &lt;- seuratFb$clusterName
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;0&quot; )] &lt;- &quot;PeriFb1&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;1&quot; )] &lt;- &quot;PDGFRAhigh Fb1&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;2&quot; )] &lt;- &quot;PDGFRAlow Fb1&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;3&quot; )] &lt;- &quot;PeriFb2&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;4&quot; )] &lt;- &quot;VSMC&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;5&quot; )] &lt;- &quot;PDGFRAlow Fb2&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;6&quot; )] &lt;- &quot;PDGFRAhigh Fb2&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;7&quot; )] &lt;- &quot;CD34 high Fb&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;8&quot; )] &lt;- &quot;PDGFRAlow VCAN high&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;9&quot; )] &lt;- &quot;PI16 high Fb&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.3 %in% &quot;10&quot; )] &lt;- &quot;PDGFRAlow CD81high&quot;
           

  
colFb &lt;- c(&quot;#e5f5f9&quot;, &quot;#99d8c9&quot;, &quot;#2ca25f&quot;, &quot;#ffffb2&quot;, &quot;#fed976&quot;, &quot;#feb24c&quot;, &quot;#fdbb84&quot;,&quot;#fd8d3c&quot;, &quot;#ef6548&quot;, &quot;#b30000&quot;, &quot;#7f0000&quot;)


names(colFb) &lt;- c(&quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;, &quot;PDGFRAhigh Fb1&quot;, &quot;PDGFRAhigh Fb2&quot;, &quot;PDGFRAlow Fb1&quot;, &quot;PDGFRAlow Fb2&quot;, &quot;PDGFRAlow VCAN high&quot;, &quot;PDGFRAlow CD81high&quot;,  &quot;CD34 high Fb&quot;, &quot;PI16 high Fb&quot;)
  
colpat_sub&lt;- c(&quot;#355C7D&quot;, &quot;#dfc27d&quot;)

names(colpat_sub) &lt;- c(&quot;DonorHeart&quot;, &quot;Sarcoidosis&quot;)

seuratFb$clusterName &lt;- factor(seuratFb$clusterName, levels = c(&quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;, &quot;PDGFRAhigh Fb1&quot;, &quot;PDGFRAhigh Fb2&quot;, &quot;PDGFRAlow Fb1&quot;, &quot;PDGFRAlow Fb2&quot;, &quot;PDGFRAlow VCAN high&quot;, &quot;PDGFRAlow CD81high&quot;,  &quot;CD34 high Fb&quot;, &quot;PI16 high Fb&quot;))


Idents(seuratFb) &lt;- seuratFb$clusterName
DimPlot(seuratFb, reduction = &quot;umap&quot;, cols = colFb, pt.size = 0.1, label =F)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Idents(seuratFb) &lt;- seuratFb$clusterName
#DimPlot(seuratFb, reduction = &quot;umap&quot;, cols = colFb, pt.size = 0.1, label =T, repel = T)</code></pre>
</div>
<div id="dotplot-with-marker-genes" class="section level2">
<h2>Dotplot with marker genes</h2>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="dotplot-with-genes-from-nature-paper-il33-feedforward"
class="section level2">
<h2>Dotplot with Genes from Nature paper IL33 feedforward</h2>
<pre class="r"><code>gene_list &lt;- rev(c(&quot;IL33&quot;,&quot;LIFR&quot;, &quot;OSMR&quot;, &quot;IL1R1&quot;,  &quot;IL6ST&quot;, &quot;IL1RAP&quot;, &quot;CCL11&quot;, &quot;IL1B&quot;, &quot;OSM&quot;, &quot;IL1RL1&quot;))

seuratM$clusterName &lt;- factor(seuratM$clusterName, levels = c(&quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;, &quot;PDGFRAhigh Fb1&quot;, &quot;PDGFRAhigh Fb2&quot;, &quot;PDGFRAlow Fb1&quot;, &quot;PDGFRAlow Fb2&quot;, &quot;PDGFRAlow VCAN high&quot;, &quot;PDGFRAlow CD81high&quot;,  &quot;CD34 high Fb&quot;, &quot;PI16 high Fb&quot;))


DotPlot(data_conv, features = gene_list, group.by= &quot;clusterName&quot;) + RotatedAxis() + scale_color_viridis(option=&quot;T&quot;) + coord_flip() + ggtitle(&quot;Celltype assignment&quot; + theme(axis.text.y = element_text(size = 10)))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="violin-plots-vlnplots-il33-and-osmr" class="section level2">
<h2>Violin Plots (VlnPlots) Il33 and OSMR</h2>
<pre class="r"><code>VlnPlot(seuratFb, features = &quot;ENSG00000137033.IL33&quot;, pt.size = 0.5, group.by =&quot;clusterName&quot;) + scale_fill_manual(values = colFb) + ggtitle (&quot;IL33&quot;)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratFb, features = &quot;ENSG00000145623.OSMR&quot;, pt.size = 0.5, group.by =&quot;clusterName&quot;) + scale_fill_manual(values = colFb) + ggtitle (&quot;OSMR&quot;)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratFb, features = &quot;ENSG00000137033.IL33&quot;, pt.size = 0.5, group.by =&quot;clusterName&quot;) +
  scale_fill_manual(values = colFb) + ggtitle (&quot;IL33&quot;)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-13-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratFb, features = &quot;ENSG00000113594.LIFR&quot;, pt.size = 0.5, group.by =&quot;clusterName&quot;) +
  scale_fill_manual(values = colFb) + ggtitle (&quot;LIFR&quot;)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-13-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratFb, features = &quot;ENSG00000134352.IL6ST&quot;, pt.size = 0.5, group.by =&quot;clusterName&quot;) +
  scale_fill_manual(values = colFb) + ggtitle (&quot;IL6ST&quot;)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-13-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratFb, features = &quot;ENSG00000115594.IL1R1&quot;, pt.size = 0.5, group.by =&quot;clusterName&quot;) +
  scale_fill_manual(values = colFb) + ggtitle (&quot;IL1R1&quot;)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-13-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratFb, features = &quot;ENSG00000196083.IL1RAP&quot;, pt.size = 0.5, group.by =&quot;clusterName&quot;) +
  scale_fill_manual(values = colFb) + ggtitle (&quot;IL1RAP&quot;)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-13-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratFb, features = &quot;ENSG00000172156.CCL11&quot;, pt.size = 0.5, group.by =&quot;clusterName&quot;) +
  scale_fill_manual(values = colFb) + ggtitle (&quot;CCL11&quot;)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-13-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>VlnPlots BMP Family</p>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratFb)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID=c(&quot;HGF&quot;, &quot;MET&quot;, &quot;GREM1&quot;, &quot;GREM2&quot;, &quot;BMPR1A&quot;, &quot;BMPR2&quot;, &quot;BMP2K&quot;, &quot;BMP8A&quot;, &quot;BMP1&quot;, &quot;BMP6&quot;, &quot;BMP2&quot;, &quot;BMP4&quot;, &quot;BMP5&quot;, &quot;BMP7&quot;)) %&gt;% 
  left_join(., genes, by=&quot;geneID&quot;)

Idents(seuratFb) &lt;- seuratFb$clusterName
pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- VlnPlot( object = seuratFb, features = x, pt.size = 0.5, cols=colFb)
  plot(p)
})</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-14-14.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="relative-abundance-of-fibroblasts-per-condition"
class="section level2">
<h2>Relative abundance of fibroblasts per condition</h2>
<pre class="r"><code>datList &lt;- NULL
for(con in unique(seuratFb$pat_sub)){
  seuratSub &lt;- subset(seuratFb, pat_sub==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(diseaseCond=con)
  datList[[con]] &lt;- dat_con
  
}</code></pre>
<pre><code>[1] 44729 19792
[1] 44729  4948</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

order_patsub &lt;- c(&quot;DonorHeart&quot;, &quot;Sarcoidosis&quot;)

ggbarplot(dat_all, x = &quot;diseaseCond&quot;, y = &quot;percent&quot;, fill = &quot;Var1&quot;, palette = colFb, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, xlab = &quot;condition&quot;, ylab = &quot;frequency&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_x_discrete(limits = order_patsub)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="relative-abundance-of-fibroblasts-per-patient"
class="section level2">
<h2>Relative abundance of fibroblasts per patient</h2>
<pre class="r"><code>datList &lt;- NULL
for(con in unique(seuratFb$patient)){
  seuratFbSub &lt;- subset(seuratFb, patient==con)
  print(dim(seuratFbSub))
  dat_con &lt;- as.data.frame(table(seuratFbSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratFbSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 44729  1880
[1] 44729   668
[1] 44729  1621
[1] 44729  2067
[1] 44729  2181
[1] 44729  2910
[1] 44729  4856
[1] 44729   889
[1] 44729  1049
[1] 44729   403
[1] 44729  1268
[1] 44729  1978
[1] 44729   636
[1] 44729    84
[1] 44729  1154
[1] 44729  1096</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

#head(dat_all)

patient_order &lt;- c(&quot;Donor Heart 01&quot;, &quot;Donor Heart 02&quot;, &quot;Donor Heart 03&quot;, &quot;Donor Heart 04&quot;, &quot;Donor Heart 05&quot;, &quot;Donor Heart 06&quot;, &quot;Donor Heart 07&quot;, &quot;Donor Heart 08&quot;, &quot;Donor Heart 09&quot;, &quot;Donor Heart 10&quot;, &quot;Donor Heart 11&quot;, &quot;Sarcoidosis 01&quot;, &quot;Sarcoidosis 02&quot;, &quot;Sarcoidosis 03&quot;, &quot;Sarcoidosis 04&quot;, &quot;Sarcoidosis 05&quot;)

#View(dat_all)

#write.table(dat_all, file= &quot;~/Documents/01 Projects/00 snRNAseq/01 Myocarditis snRNAseq/docs/Markers&quot;, sep = &quot;\t&quot;,quote=F,row.names=T,col.names=T)


dat_all$patient &lt;- factor(dat_all$patient, levels = patient_order)

ggbarplot(dat_all, x= &quot;patient&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;, palette = colFb)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fb-counts" class="section level2">
<h2>Fb Counts</h2>
<pre class="r"><code>## total Fb count for all conditions
table(seuratFb$orig.ident)</code></pre>
<pre><code>
      
24740 </code></pre>
<pre class="r"><code>## Clusters per condition
table(seuratFb$clusterName, seuratFb$pat_sub)</code></pre>
<pre><code>                     
                      DonorHeart Sarcoidosis
  PeriFb1                   6574         967
  PeriFb2                   2436         273
  VSMC                      1253         333
  PDGFRAhigh Fb1            3699          92
  PDGFRAhigh Fb2            1075           6
  PDGFRAlow Fb1             3592          67
  PDGFRAlow Fb2               40        1462
  PDGFRAlow VCAN high         64         726
  PDGFRAlow CD81high          13         440
  CD34 high Fb              1015           7
  PI16 high Fb                31         575</code></pre>
</div>
<div id="fb-subtype-relative-abundance-per-condition"
class="section level2">
<h2>Fb subtype relative abundance per condition</h2>
<pre class="r"><code>seuratFb$pat_sub &lt;- factor(seuratFb$pat_sub, levels = c(&quot;DonorHeart&quot;, &quot;Sarcoidosis&quot;))


colpat_sub &lt;- c(&quot;DonorHeart&quot; = &quot;#355C7D&quot;, &quot;Sarcoidosis&quot; = &quot;#dfc27d&quot;)


dat_counts &lt;- seuratFb@meta.data %&gt;%
  group_by(patient_pat_sub, pat_sub, clusterName) %&gt;%
  summarise(cnt = n(), .groups = &quot;drop&quot;)

dat_totals &lt;- seuratFb@meta.data %&gt;%
  group_by(patient_pat_sub) %&gt;%
  summarise(total_cells = n(), .groups = &quot;drop&quot;)

dat_rel &lt;- dat_counts %&gt;%
  left_join(dat_totals, by = &quot;patient_pat_sub&quot;) %&gt;%
  mutate(rel_abundance = cnt / total_cells)


pvals &lt;- dat_rel %&gt;%
  group_by(clusterName) %&gt;%
  summarise(p_val = wilcox.test(rel_abundance ~ pat_sub)$p.value) %&gt;%
  mutate(p_label = sprintf(&quot;p = %.3f&quot;, p_val))


y_max &lt;- dat_rel %&gt;% 
  group_by(clusterName) %&gt;%
  summarise(y_pos = max(rel_abundance))


pvals &lt;- left_join(pvals, y_max, by = &quot;clusterName&quot;)


ggbarplot(dat_rel,
          x = &quot;clusterName&quot;,
          y = &quot;rel_abundance&quot;,
          fill = &quot;pat_sub&quot;,
          palette = colpat_sub,
          add = c(&quot;mean_se&quot;),
          position = position_dodge(0.8),
          xlab = &quot;Cluster (Cell type)&quot;,
          ylab = &quot;Relative abundance&quot;,
          legend.title = &quot;Condition&quot;) +
  
  theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  
  ggrepel::geom_text_repel(data = pvals, aes(x = clusterName, y = y_pos * 1.05, label = p_label), inherit.aes = FALSE,
size = 3, direction = &quot;y&quot;,nudge_y = 0.01, segment.size = 0.2, box.padding = 0.5, point.padding = 0.5)</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fb-feature-plots" class="section level2">
<h2>Fb Feature Plots</h2>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-14.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-16.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-18.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-20.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-21.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-22.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-23.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-24.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-25.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-26.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-27.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-28.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-29.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-30.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-31.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-32.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-33.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-34.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-35.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-36.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-37.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-38.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-39.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-40.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-41.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-42.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-43.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-44.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-45.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-20-46.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="featureplots-il33-axis-projected-onto-gsea"
class="section level2">
<h2>FeaturePlots IL33 axis projected onto GSEA</h2>
<pre class="r"><code>FeaturePlot(seuratFb, features = &quot;ENSG00000145623.OSMR&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratFb, features = &quot;ENSG00000115602.IL1RL1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratFb, features = &quot;ENSG00000137033.IL33&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-21-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratFb, features = &quot;ENSG00000145623.OSMR&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-21-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratFb, features = &quot;ENSG00000113594.LIFR&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-21-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratFb, features = &quot;ENSG00000134352.IL6ST&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-21-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratFb, features = &quot;ENSG00000115594.IL1R1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-21-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratFb, features = &quot;ENSG00000196083.IL1RAP&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-21-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratFb, features = get_full_gene_name(&quot;CCL11&quot;,seuratM) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-21-9.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="feature-plots-of-bmp-family" class="section level2">
<h2>Feature Plots of BMP Family</h2>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratFb)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID=c(&quot;HGF&quot;, &quot;MET&quot;, &quot;GREM1&quot;, &quot;GREM2&quot;, &quot;BMPR1A&quot;, &quot;BMPR2&quot;, &quot;BMP2K&quot;, &quot;BMP8A&quot;, &quot;BMP1&quot;, &quot;BMP6&quot;, &quot;BMP2&quot;, &quot;BMP4&quot;, &quot;BMP5&quot;, &quot;BMP7&quot;)) %&gt;% 
  left_join(., genes, by=&quot;geneID&quot;)

pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- FeaturePlot(seuratFb, features = x, 
                   reduction = &quot;umap&quot;, pt.size = 0.1, 
                   cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), 
                   raster = FALSE) +
    theme(legend.position=&quot;right&quot;)
  plot(p)
})</code></pre>
<p><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fibroblast_GSEA_Nature_Validation.Rmd/unnamed-chunk-22-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_2.0.4              future_1.67.0               workflowr_1.7.2            
 [4] pathview_1.46.0             wordcloud_2.6               textshaping_1.0.3          
 [7] EnhancedVolcano_1.24.0      ggrepel_0.9.6               RColorBrewer_1.1-3         
[10] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
[13] ggupset_0.4.1               gridExtra_2.3               DOSE_4.0.1                 
[16] enrichplot_1.26.6           msigdbr_25.1.1              org.Hs.eg.db_3.20.0        
[19] AnnotationDbi_1.68.0        clusterProfiler_4.14.6      multtest_2.62.0            
[22] metap_1.12                  scater_1.34.1               scuttle_1.16.0             
[25] destiny_3.20.0              circlize_0.4.16             muscat_1.20.0              
[28] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[31] forcats_1.0.1               stringr_1.5.2               purrr_1.1.0                
[34] readr_2.1.5                 tidyr_1.3.1                 tibble_3.3.0               
[37] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.28.1
[40] SummarizedExperiment_1.36.0 Biobase_2.66.0              GenomicRanges_1.58.0       
[43] GenomeInfoDb_1.42.3         IRanges_2.40.1              S4Vectors_0.44.0           
[46] BiocGenerics_0.52.0         MatrixGenerics_1.18.1       matrixStats_1.5.0          
[49] pheatmap_1.0.13             ggpubr_0.6.1                ggplot2_4.0.0              
[52] Seurat_5.3.0                SeuratObject_5.2.0          sp_2.2-0                   

loaded via a namespace (and not attached):
  [1] graph_1.84.1             igraph_2.1.4             ica_1.0-3               
  [4] plotly_4.11.0            Formula_1.2-5            zlibbioc_1.52.0         
  [7] tidyselect_1.2.1         bit_4.6.0                doParallel_1.0.17       
 [10] clue_0.3-66              lattice_0.22-7           rjson_0.2.23            
 [13] blob_1.2.4               S4Arrays_1.6.0           pbkrtest_0.5.5          
 [16] parallel_4.4.2           png_0.1-8                plotrix_3.8-4           
 [19] cli_3.6.5                ggplotify_0.1.3          goftest_1.2-3           
 [22] VIM_6.2.6                variancePartition_1.36.3 BiocNeighbors_2.0.1     
 [25] uwot_0.2.3               curl_7.0.0               mime_0.13               
 [28] evaluate_1.0.5           tidytree_0.4.6           ComplexHeatmap_2.22.0   
 [31] stringi_1.8.7            backports_1.5.0          XML_3.99-0.19           
 [34] lmerTest_3.1-3           qqconf_1.3.2             httpuv_1.6.16           
 [37] rappdirs_0.3.3           splines_4.4.2            sctransform_0.4.2       
 [40] ggbeeswarm_0.7.2         DBI_1.2.3                jquerylib_0.1.4         
 [43] smoother_1.3             withr_3.0.2              git2r_0.36.2            
 [46] corpcor_1.6.10           systemfonts_1.2.3        reformulas_0.4.1        
 [49] class_7.3-23             rprojroot_2.1.1          lmtest_0.9-40           
 [52] formatR_1.14             htmlwidgets_1.6.4        fs_1.6.6                
 [55] labeling_0.4.3           fANCOVA_0.6-1            SparseArray_1.6.2       
 [58] DESeq2_1.46.0            ranger_0.17.0            DEoptimR_1.1-4          
 [61] reticulate_1.43.0        hexbin_1.28.5            zoo_1.8-14              
 [64] XVector_0.46.0           knitr_1.50               ggplot.multistats_1.0.1 
 [67] UCSC.utils_1.2.0         RhpcBLASctl_0.23-42      timechange_0.3.0        
 [70] foreach_1.5.2            patchwork_1.3.2          caTools_1.18.3          
 [73] data.table_1.17.8        ggtree_3.14.0            R.oo_1.27.1             
 [76] RSpectra_0.16-2          irlba_2.3.5.1            ggrastr_1.0.2           
 [79] fastDummies_1.7.5        gridGraphics_0.5-1       lazyeval_0.2.2          
 [82] yaml_2.3.10              survival_3.8-3           scattermore_1.2         
 [85] crayon_1.5.3             RcppAnnoy_0.0.22         Rgraphviz_2.50.0        
 [88] progressr_0.16.0         later_1.4.4              ggridges_0.5.7          
 [91] codetools_0.2-20         GlobalOptions_0.1.2      aod_1.3.3               
 [94] KEGGREST_1.46.0          Rtsne_0.17               shape_1.4.6.1           
 [97] limma_3.62.2             estimability_1.5.1       pkgconfig_2.0.3         
[100] KEGGgraph_1.66.0         TMB_1.9.17               spatstat.univar_3.1-4   
[103] mathjaxr_1.8-0           getPass_0.2-4            EnvStats_3.1.0          
[106] aplot_0.2.9              scatterplot3d_0.3-44     spatstat.sparse_3.1-0   
[109] ape_5.8-1                xtable_1.8-4             car_3.1-3               
[112] plyr_1.8.9               httr_1.4.7               rbibutils_2.3           
[115] tools_4.4.2              globals_0.18.0           beeswarm_0.4.0          
[118] broom_1.0.10             nlme_3.1-168             lambda.r_1.2.4          
[121] assertthat_0.2.1         lme4_1.1-37              digest_0.6.37           
[124] numDeriv_2016.8-1.1      Matrix_1.7-4             farver_2.1.2            
[127] tzdb_0.5.0               remaCor_0.0.20           reshape2_1.4.4          
[130] yulab.utils_0.2.1        glue_1.8.0               cachem_1.1.0            
[133] polyclip_1.10-7          generics_0.1.4           Biostrings_2.74.1       
[136] mvtnorm_1.3-3            presto_1.0.0             parallelly_1.45.1       
[139] mnormt_2.1.1             statmod_1.5.0            RcppHNSW_0.6.0          
[142] ScaledMatrix_1.14.0      carData_3.0-5            minqa_1.2.8             
[145] pbapply_1.7-4            spam_2.11-1              gson_0.1.0              
[148] gtools_3.9.5             ggsignif_0.6.4           RcppEigen_0.3.4.0.2     
[151] shiny_1.11.1             GenomeInfoDbData_1.2.13  glmmTMB_1.1.12          
[154] R.utils_2.13.0           RCurl_1.98-1.17          memoise_2.0.1           
[157] rmarkdown_2.29           scales_1.4.0             R.methodsS3_1.8.2       
[160] RANN_2.6.2               spatstat.data_3.1-8      rstudioapi_0.17.1       
[163] whisker_0.4.1            cluster_2.1.8.1          mutoss_0.1-13           
[166] spatstat.utils_3.2-0     hms_1.1.3                fitdistrplus_1.2-4      
[169] cowplot_1.2.0            colorspace_2.1-2         rlang_1.1.6             
[172] xts_0.14.1               dotCall64_1.2            ggtangle_0.0.7          
[175] laeken_0.5.3             mgcv_1.9-3               xfun_0.53               
[178] coda_0.19-4.1            e1071_1.7-16             TH.data_1.1-4           
[181] iterators_1.0.14         emmeans_1.11.2-8         abind_1.4-8             
[184] GOSemSim_2.32.0          treeio_1.30.0            ps_1.9.1                
[187] futile.options_1.0.1     bitops_1.0-9             Rdpack_2.6.4            
[190] promises_1.3.3           RSQLite_2.4.3            qvalue_2.38.0           
[193] sandwich_3.1-1           fgsea_1.32.4             DelayedArray_0.32.0     
[196] proxy_0.4-27             GO.db_3.20.0             compiler_4.4.2          
[199] prettyunits_1.2.0        boot_1.3-32              beachmat_2.22.0         
[202] listenv_0.9.1            Rcpp_1.1.0               edgeR_4.4.2             
[205] BiocSingular_1.22.0      tensor_1.5.1             MASS_7.3-65             
[208] progress_1.2.3           BiocParallel_1.40.2      babelgene_22.9          
[211] spatstat.random_3.4-2    R6_2.6.1                 fastmap_1.2.0           
[214] multcomp_1.4-28          fastmatch_1.1-6          rstatix_0.7.2           
[217] vipor_0.4.7              TTR_0.24.4               ROCR_1.0-11             
[220] TFisher_0.2.0            rsvd_1.0.5               vcd_1.4-13              
[223] nnet_7.3-20              gtable_0.3.6             KernSmooth_2.23-26      
[226] miniUI_0.1.2             deldir_2.0-4             htmltools_0.5.8.1       
[229] ggthemes_5.1.0           bit64_4.6.0-1            spatstat.explore_3.5-3  
[232] lifecycle_1.0.4          blme_1.0-6               S7_0.2.0                
[235] processx_3.8.6           callr_3.7.6              nloptr_2.2.1            
[238] sass_0.4.10              vctrs_0.6.5              robustbase_0.99-6       
[241] spatstat.geom_3.6-0      sn_2.1.1                 ggfun_0.2.0             
[244] future.apply_1.20.0      bslib_0.9.0              pillar_1.11.1           
[247] gplots_3.2.0             pcaMethods_1.98.0        locfit_1.5-9.12         
[250] jsonlite_2.0.0           GetoptLong_1.0.5        </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Sun Oct  5 18:04:06 2025&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_2.0.4              future_1.67.0               workflowr_1.7.2            
 [4] pathview_1.46.0             wordcloud_2.6               textshaping_1.0.3          
 [7] EnhancedVolcano_1.24.0      ggrepel_0.9.6               RColorBrewer_1.1-3         
[10] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
[13] ggupset_0.4.1               gridExtra_2.3               DOSE_4.0.1                 
[16] enrichplot_1.26.6           msigdbr_25.1.1              org.Hs.eg.db_3.20.0        
[19] AnnotationDbi_1.68.0        clusterProfiler_4.14.6      multtest_2.62.0            
[22] metap_1.12                  scater_1.34.1               scuttle_1.16.0             
[25] destiny_3.20.0              circlize_0.4.16             muscat_1.20.0              
[28] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[31] forcats_1.0.1               stringr_1.5.2               purrr_1.1.0                
[34] readr_2.1.5                 tidyr_1.3.1                 tibble_3.3.0               
[37] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.28.1
[40] SummarizedExperiment_1.36.0 Biobase_2.66.0              GenomicRanges_1.58.0       
[43] GenomeInfoDb_1.42.3         IRanges_2.40.1              S4Vectors_0.44.0           
[46] BiocGenerics_0.52.0         MatrixGenerics_1.18.1       matrixStats_1.5.0          
[49] pheatmap_1.0.13             ggpubr_0.6.1                ggplot2_4.0.0              
[52] Seurat_5.3.0                SeuratObject_5.2.0          sp_2.2-0                   

loaded via a namespace (and not attached):
  [1] graph_1.84.1             igraph_2.1.4             ica_1.0-3               
  [4] plotly_4.11.0            Formula_1.2-5            zlibbioc_1.52.0         
  [7] tidyselect_1.2.1         bit_4.6.0                doParallel_1.0.17       
 [10] clue_0.3-66              lattice_0.22-7           rjson_0.2.23            
 [13] blob_1.2.4               S4Arrays_1.6.0           pbkrtest_0.5.5          
 [16] parallel_4.4.2           png_0.1-8                plotrix_3.8-4           
 [19] cli_3.6.5                ggplotify_0.1.3          goftest_1.2-3           
 [22] VIM_6.2.6                variancePartition_1.36.3 BiocNeighbors_2.0.1     
 [25] uwot_0.2.3               curl_7.0.0               mime_0.13               
 [28] evaluate_1.0.5           tidytree_0.4.6           ComplexHeatmap_2.22.0   
 [31] stringi_1.8.7            backports_1.5.0          XML_3.99-0.19           
 [34] lmerTest_3.1-3           qqconf_1.3.2             httpuv_1.6.16           
 [37] rappdirs_0.3.3           splines_4.4.2            sctransform_0.4.2       
 [40] ggbeeswarm_0.7.2         DBI_1.2.3                jquerylib_0.1.4         
 [43] smoother_1.3             withr_3.0.2              git2r_0.36.2            
 [46] corpcor_1.6.10           systemfonts_1.2.3        reformulas_0.4.1        
 [49] class_7.3-23             rprojroot_2.1.1          lmtest_0.9-40           
 [52] formatR_1.14             htmlwidgets_1.6.4        fs_1.6.6                
 [55] labeling_0.4.3           fANCOVA_0.6-1            SparseArray_1.6.2       
 [58] DESeq2_1.46.0            ranger_0.17.0            DEoptimR_1.1-4          
 [61] reticulate_1.43.0        hexbin_1.28.5            zoo_1.8-14              
 [64] XVector_0.46.0           knitr_1.50               ggplot.multistats_1.0.1 
 [67] UCSC.utils_1.2.0         RhpcBLASctl_0.23-42      timechange_0.3.0        
 [70] foreach_1.5.2            patchwork_1.3.2          caTools_1.18.3          
 [73] data.table_1.17.8        ggtree_3.14.0            R.oo_1.27.1             
 [76] RSpectra_0.16-2          irlba_2.3.5.1            ggrastr_1.0.2           
 [79] fastDummies_1.7.5        gridGraphics_0.5-1       lazyeval_0.2.2          
 [82] yaml_2.3.10              survival_3.8-3           scattermore_1.2         
 [85] crayon_1.5.3             RcppAnnoy_0.0.22         Rgraphviz_2.50.0        
 [88] progressr_0.16.0         later_1.4.4              ggridges_0.5.7          
 [91] codetools_0.2-20         GlobalOptions_0.1.2      aod_1.3.3               
 [94] KEGGREST_1.46.0          Rtsne_0.17               shape_1.4.6.1           
 [97] limma_3.62.2             estimability_1.5.1       pkgconfig_2.0.3         
[100] KEGGgraph_1.66.0         TMB_1.9.17               spatstat.univar_3.1-4   
[103] mathjaxr_1.8-0           getPass_0.2-4            EnvStats_3.1.0          
[106] aplot_0.2.9              scatterplot3d_0.3-44     spatstat.sparse_3.1-0   
[109] ape_5.8-1                xtable_1.8-4             car_3.1-3               
[112] plyr_1.8.9               httr_1.4.7               rbibutils_2.3           
[115] tools_4.4.2              globals_0.18.0           beeswarm_0.4.0          
[118] broom_1.0.10             nlme_3.1-168             lambda.r_1.2.4          
[121] assertthat_0.2.1         lme4_1.1-37              digest_0.6.37           
[124] numDeriv_2016.8-1.1      Matrix_1.7-4             farver_2.1.2            
[127] tzdb_0.5.0               remaCor_0.0.20           reshape2_1.4.4          
[130] yulab.utils_0.2.1        glue_1.8.0               cachem_1.1.0            
[133] polyclip_1.10-7          generics_0.1.4           Biostrings_2.74.1       
[136] mvtnorm_1.3-3            presto_1.0.0             parallelly_1.45.1       
[139] mnormt_2.1.1             statmod_1.5.0            RcppHNSW_0.6.0          
[142] ScaledMatrix_1.14.0      carData_3.0-5            minqa_1.2.8             
[145] pbapply_1.7-4            spam_2.11-1              gson_0.1.0              
[148] gtools_3.9.5             ggsignif_0.6.4           RcppEigen_0.3.4.0.2     
[151] shiny_1.11.1             GenomeInfoDbData_1.2.13  glmmTMB_1.1.12          
[154] R.utils_2.13.0           RCurl_1.98-1.17          memoise_2.0.1           
[157] rmarkdown_2.29           scales_1.4.0             R.methodsS3_1.8.2       
[160] RANN_2.6.2               spatstat.data_3.1-8      rstudioapi_0.17.1       
[163] whisker_0.4.1            cluster_2.1.8.1          mutoss_0.1-13           
[166] spatstat.utils_3.2-0     hms_1.1.3                fitdistrplus_1.2-4      
[169] cowplot_1.2.0            colorspace_2.1-2         rlang_1.1.6             
[172] xts_0.14.1               dotCall64_1.2            ggtangle_0.0.7          
[175] laeken_0.5.3             mgcv_1.9-3               xfun_0.53               
[178] coda_0.19-4.1            e1071_1.7-16             TH.data_1.1-4           
[181] iterators_1.0.14         emmeans_1.11.2-8         abind_1.4-8             
[184] GOSemSim_2.32.0          treeio_1.30.0            ps_1.9.1                
[187] futile.options_1.0.1     bitops_1.0-9             Rdpack_2.6.4            
[190] promises_1.3.3           RSQLite_2.4.3            qvalue_2.38.0           
[193] sandwich_3.1-1           fgsea_1.32.4             DelayedArray_0.32.0     
[196] proxy_0.4-27             GO.db_3.20.0             compiler_4.4.2          
[199] prettyunits_1.2.0        boot_1.3-32              beachmat_2.22.0         
[202] listenv_0.9.1            Rcpp_1.1.0               edgeR_4.4.2             
[205] BiocSingular_1.22.0      tensor_1.5.1             MASS_7.3-65             
[208] progress_1.2.3           BiocParallel_1.40.2      babelgene_22.9          
[211] spatstat.random_3.4-2    R6_2.6.1                 fastmap_1.2.0           
[214] multcomp_1.4-28          fastmatch_1.1-6          rstatix_0.7.2           
[217] vipor_0.4.7              TTR_0.24.4               ROCR_1.0-11             
[220] TFisher_0.2.0            rsvd_1.0.5               vcd_1.4-13              
[223] nnet_7.3-20              gtable_0.3.6             KernSmooth_2.23-26      
[226] miniUI_0.1.2             deldir_2.0-4             htmltools_0.5.8.1       
[229] ggthemes_5.1.0           bit64_4.6.0-1            spatstat.explore_3.5-3  
[232] lifecycle_1.0.4          blme_1.0-6               S7_0.2.0                
[235] processx_3.8.6           callr_3.7.6              nloptr_2.2.1            
[238] sass_0.4.10              vctrs_0.6.5              robustbase_0.99-6       
[241] spatstat.geom_3.6-0      sn_2.1.1                 ggfun_0.2.0             
[244] future.apply_1.20.0      bslib_0.9.0              pillar_1.11.1           
[247] gplots_3.2.0             pcaMethods_1.98.0        locfit_1.5-9.12         
[250] jsonlite_2.0.0           GetoptLong_1.0.5        </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
