<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Payne" />

<meta name="date" content="2025-06-05" />

<title>Sarcoidosis Validation IL-33 - Oncostatin-M axis accross all cell clusters</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiac Sarcoidosis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Sarcoidosis Validation IL-33 - Oncostatin-M
axis accross all cell clusters</h1>
<h4 class="author">Emily Payne</h4>
<h4 class="date">2025-06-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-05
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiac Sarcoidosis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20251004code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20251004)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20251004code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20251004)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Documents/Github/Cardiac
Sarcoidosis/data/Merged/Sarcoidosis_vs_DonorHeart_v2.rds
</td>
<td style="text-align:left;">
data/Merged/Sarcoidosis_vs_DonorHeart_v2.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong6ae5a3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 6ae5a3a
</a>
</p>
</div>
<div id="strongRepositoryversionstrong6ae5a3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 6ae5a3a.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    R/
    Ignored:    analysis/.DS_Store
    Ignored:    data/Merged/

Untracked files:
    Untracked:  .gitignore.gitignore
    Untracked:  Fibroblast-GSEA-Nature-Validation_files/
    Untracked:  analysis/Cellchat_Sarcoidosis_v3.Rmd
    Untracked:  analysis/Fibroblast_GSEA_Nature_Validation.Rmd
    Untracked:  analysis/Sarcoidosis_ClusterAnnotations.Rmd
    Untracked:  analysis/Sarcoidosis_GREM2.Rmd
    Untracked:  analysis/Sarcoidosis_Subanalysis.Rmd
    Untracked:  analysis/Sarcoidosis_allClusters_Validation_Nature.Rmd
    Untracked:  analysis/old files/
    Untracked:  code/gitignore

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>suppressPackageStartupMessages({
  library(Seurat)
  library(ggpubr)
  library(pheatmap)
  library(SingleCellExperiment)
  library(dplyr)
  library(tidyverse)
  library(viridis)
  library(muscat)
  library(circlize)
  library(destiny)
  library(scater)
  library(metap)
  library(multtest)
  library(clusterProfiler)
  library(org.Hs.eg.db)
  library(msigdbr)
  library(enrichplot)
  library(DOSE)
  library(grid)
  library(gridExtra)
  library(ggupset)
  library(VennDiagram)
  library(NCmisc)
  library(RColorBrewer)
  library(EnhancedVolcano)
  library(textshaping)
  library(wordcloud)
  library(pathview)
  library(workflowr)
  library(future)
  library(tibble)
  library(tidyr)
library(circlize)
library(magrittr)
})</code></pre>
<pre class="r"><code>seuratM &lt;- readRDS(&quot;~/Documents/Github/Cardiac Sarcoidosis/data/Merged/Sarcoidosis_vs_DonorHeart_v2.rds&quot;)
&quot;data/Merged/Sarcoidosis_vs_DonorHeart_v2.rds&quot;</code></pre>
<pre><code>[1] &quot;data/Merged/Sarcoidosis_vs_DonorHeart_v2.rds&quot;</code></pre>
<pre><code>
      
49601 </code></pre>
<div id="umap" class="section level2">
<h2>UMAP</h2>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-with-marker-genes" class="section level1">
<h1>Heatmap with Marker Genes</h1>
<pre><code> [1] &quot;CM&quot;          &quot;Fb1&quot;         &quot;PeriFb&quot;      &quot;VSMC&quot;        &quot;BEC1&quot;        &quot;BEC2&quot;       
 [7] &quot;NC1&quot;         &quot;TRM&quot;         &quot;inflMph&quot;     &quot;Lymphocytes&quot;</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="heatmap-with-il33-osm-axis" class="section level2">
<h2>Heatmap with IL33-OSM axis</h2>
<pre><code> [1] &quot;CM&quot;          &quot;Fb1&quot;         &quot;PeriFb&quot;      &quot;VSMC&quot;        &quot;BEC1&quot;        &quot;BEC2&quot;       
 [7] &quot;NC1&quot;         &quot;TRM&quot;         &quot;inflMph&quot;     &quot;Lymphocytes&quot;</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="dotplot-with-genes-il33-osm-axis" class="section level2">
<h2>Dotplot with genes: IL33-OSM axis</h2>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" />
## Feature Plots with genes: IL33-OSM axis</p>
<pre><code>[1] &quot;ENSG00000249740.OSMR-AS1&quot; &quot;ENSG00000145623.OSMR&quot;     &quot;ENSG00000185716.MOSMO&quot;   
[4] &quot;ENSG00000260552.COSMOC&quot;   &quot;ENSG00000099985.OSM&quot;     </code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-9-11.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="vln-plots-genes-involved-in-il33-osm-axis-per-cluster-accross-all-conditions"
class="section level2">
<h2>Vln Plots genes involved in IL33-OSM axis per cluster accross all
conditions</h2>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-10-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-10-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-10-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-10-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-10-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-10-9.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="violin-plots-with-genes-involved-in-the-il133-osm-axis-by-condition"
class="section level2">
<h2>Violin Plots with genes involved in the IL133-OSM axis by
condition</h2>
<pre class="r"><code> #c(&quot;ENSG00000125538.IL1B&quot;, &quot;ENSG00000099985.OSM&quot;, &quot;ENSG00000115602.IL1RL1&quot;, &quot;ENSG00000137033.IL33&quot;, #&quot;ENSG00000113594.LIFR&quot;, &quot;ENSG00000134352.IL6ST&quot;, &quot;ENSG00000115594.IL1R1&quot;, &quot;ENSG00000196083.IL1RAP&quot;)

VlnPlot(seuratM, features = &quot;ENSG00000125538.IL1B&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#355C7D&quot;, &quot;#dfc27d&quot;)) + ggtitle (&quot;IL1B&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratM, features = &quot;ENSG00000099985.OSM&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#355C7D&quot;, &quot;#dfc27d&quot;)) + ggtitle (&quot;OSM&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratM, features = &quot;ENSG00000115602.IL1RL1&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#355C7D&quot;, &quot;#dfc27d&quot;)) + ggtitle (&quot;IL1RL1&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratM, features = &quot;ENSG00000137033.IL33&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#355C7D&quot;, &quot;#dfc27d&quot;)) + ggtitle (&quot;IL33&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-11-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratM, features = &quot;ENSG00000113594.LIFR&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#355C7D&quot;, &quot;#dfc27d&quot;)) + ggtitle (&quot;LIFR&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-11-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratM, features = &quot;ENSG00000134352.IL6ST&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#355C7D&quot;, &quot;#dfc27d&quot;)) + ggtitle (&quot;IL6ST&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-11-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratM, features = &quot;ENSG00000115594.IL1R1&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#355C7D&quot;, &quot;#dfc27d&quot;)) + ggtitle (&quot;IL1R1&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-11-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratM, features = &quot;ENSG00000196083.IL1RAP&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#355C7D&quot;, &quot;#dfc27d&quot;)) + ggtitle (&quot;IL1RAP&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-11-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratM, features = &quot;ENSG00000172156.CCL11&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#355C7D&quot;, &quot;#dfc27d&quot;)) + ggtitle (&quot;CCL11&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-11-9.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-of-genes-clusterwise-per-condition"
class="section level2">
<h2>Comparison of genes clusterwise per condition</h2>
<pre class="r"><code>colpat_sub &lt;- c(&quot;DonorHeart&quot; = &quot;#355C7D&quot;, &quot;Sarcoidosis&quot; = &quot;#dfc27d&quot;)
#colclusterName &lt;- c(&quot;Macrophages&quot; = &quot;#1f77b4&quot;, &quot;T cells&quot; = &quot;#ff7f0e&quot;, &quot;NK cells&quot; = &quot;#2ca02c&quot;)  # Adjust as needed


genes &lt;- c(&quot;ENSG00000125538.IL1B&quot;, &quot;ENSG00000099985.OSM&quot;, &quot;ENSG00000115602.IL1RL1&quot;, &quot;ENSG00000137033.IL33&quot;, &quot;ENSG00000113594.LIFR&quot;, &quot;ENSG00000134352.IL6ST&quot;, &quot;ENSG00000115594.IL1R1&quot;, &quot;ENSG00000196083.IL1RAP&quot; )


for (gene_name in genes) {
  
  
  expr &lt;- GetAssayData(seuratM, slot = &quot;data&quot;)[gene_name, ]
  
 
  df &lt;- data.frame(expression = as.numeric(expr), clusterName = seuratM$clusterName, pat_sub = seuratM$pat_sub)
  
  df$clusterName &lt;- factor(df$clusterName, levels = names(colclusterName))
  df$pat_sub &lt;- factor(df$pat_sub, levels = names(colpat_sub))
  

  pvals &lt;- df %&gt;%
    group_by(clusterName) %&gt;%
    summarise(
      p_val = wilcox.test(expression ~ pat_sub, exact = FALSE)$p.value,
      .groups = &quot;drop&quot;) %&gt;%
    mutate(p_label = sprintf(&quot;p = %.3f&quot;, p_val))
  
  y_pos &lt;- df %&gt;%
    group_by(clusterName) %&gt;%
    summarise(y = max(expression, na.rm = TRUE) * 1.05, .groups = &quot;drop&quot;)
  
  pvals &lt;- left_join(pvals, y_pos, by = &quot;clusterName&quot;)
  

 graph &lt;- ggplot(df, aes(x = clusterName, y = expression, fill = pat_sub)) +
  geom_violin(position = position_dodge(width = 0.9), trim = TRUE, alpha = 0.7) +
  geom_jitter(
    position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.9),
    size = 0.5,
    alpha = 0.6,
    color = &quot;black&quot;) +
  scale_fill_manual(values = colpat_sub) +
  theme_classic() +
  labs(title = paste(&quot;Expression of&quot;, gene_name, &quot;by Cluster and Condition&quot;),
       x = &quot;Cell Type (Cluster)&quot;,
       y = &quot;Normalized Expression&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, color = colclusterName[levels(df$clusterName)]),
    plot.title = element_text(hjust = 0.5)) + geom_text(data = pvals, aes(x = clusterName, y = y, label = p_label),
            inherit.aes = FALSE, vjust = 0, size = 3)

  
  print(graph)
}</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-12-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-12-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-12-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-12-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-12-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-12-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-ontology-for-fibroblasts-and-macrophages"
class="section level2">
<h2>Gene ontology for fibroblasts and macrophages</h2>
<pre class="r"><code>## DE Genes Fibroblasts
DEGenesFb &lt;- DEGenesClusters %&gt;% filter(cluster == &quot;Fb1&quot;)

egoFb &lt;- enrichGO(gene = unique(DEGenesFb$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

egoFb &lt;- setReadable(egoFb, OrgDb = org.Hs.eg.db)


#dotplot(egoFb, showCategory=20, title = &quot;Fibroblast G0 terms&quot;, font.size = 10)

#View(egoFb@result)

barplot(egoFb, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Fibroblast G0 terms&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes Tissue resident macrophages
DEGenesTRM &lt;- DEGenesClusters %&gt;% filter(cluster == &quot;TRM&quot;)

egoTRM &lt;- enrichGO(gene = unique(DEGenesTRM$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

egoTRM &lt;- setReadable(egoTRM, OrgDb = org.Hs.eg.db)


#dotplot(egoTRM, showCategory=20, title = &quot;Tissue resident macrophages GO terms&quot;, font.size = 10)

#View(egoTRM@result)

barplot(egoTRM, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Tissue resident macrophages GO terms&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes inflammatory macrophages
DEGenesinflMph &lt;- DEGenesClusters %&gt;% filter(cluster == &quot;inflMph&quot;)

egoMph &lt;- enrichGO(gene = unique(DEGenesinflMph$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

egoMph &lt;- setReadable(egoMph, OrgDb = org.Hs.eg.db)


#dotplot(egoMph, showCategory=20, title = &quot;Inflammatory macrophages&quot;, font.size = 10)

#View(egoMph@result)

barplot(egoMph, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Inflammatory Macrophages GO terms&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="single-cell-experiment-with-pathway-analysis-projected-on-umap"
class="section level3">
<h3>Single cell Experiment with Pathway Analysis projected on UMAP</h3>
<pre class="r"><code>sce &lt;- as.SingleCellExperiment(seuratM)
genes &lt;- data.frame(geneID=rownames(sce)) %&gt;% mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;, geneID))
pal = colorRampPalette(c(&quot;#053061&quot;, &quot;#2166ac&quot;, &quot;#f7f7f7&quot;, &quot;#f4a582&quot;,&quot;orange&quot;, &quot;#d95f0e&quot;,&quot;#b2183c&quot;))
scale_fill_gradientn(colors = pal(101), na.value = &quot;grey90&quot;)</code></pre>
<pre><code>&lt;ScaleContinuous&gt;
 Range:  
 Limits:    0 --    1</code></pre>
<pre class="r"><code>seuratDH &lt;- subset(seuratM, pat_sub == &quot;DonorHeart&quot;)
table(seuratDH$pat_sub)</code></pre>
<pre><code>
DonorHeart 
     40222 </code></pre>
<pre class="r"><code>sceDH &lt;- as.SingleCellExperiment(seuratDH)</code></pre>
<pre class="r"><code>seuratSarco &lt;- subset(seuratM, pat_sub == &quot;Sarcoidosis&quot;)
table(seuratSarco$pat_sub)</code></pre>
<pre><code>
Sarcoidosis 
       9379 </code></pre>
<pre class="r"><code>sceSarco &lt;- as.SingleCellExperiment(seuratSarco)</code></pre>
</div>
</div>
<div id="leukocyte-migration-pathway-projected-on-umap"
class="section level2">
<h2>Leukocyte migration pathway projected on UMAP</h2>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis 
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
<pre class="r"><code>## Donor Heart
sceDHSub &lt;- sceDH[which(rownames(sceDH) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceDHSub@assays@data$logcounts)))/nrow(signGenes)
sceDHSub$sign &lt;- cntMat
sceDHSub$sign2 &lt;- sceDHSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceDHSub$sign2[which(sceDHSub$sign &gt; 1)] &lt;- 1

#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceDHSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="extracellular-matrix-organisation" class="section level2">
<h2>Extracellular matrix organisation</h2>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis: ECM organisation
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
<pre class="r"><code>## Donor Heart: ECM organisation
sceDHSub &lt;- sceDH[which(rownames(sceDH) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceDHSub@assays@data$logcounts)))/nrow(signGenes)
sceDHSub$sign &lt;- cntMat
sceDHSub$sign2 &lt;- sceDHSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceDHSub$sign2[which(sceDHSub$sign &gt; 1)] &lt;- 1

#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceDHSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="response-to-interferon-beta" class="section level2">
<h2>Response to Interferon beta</h2>
<pre class="r"><code>egoFb3 &lt;- dplyr::filter(egoFb@result, egoFb@result$Description %in% c(&quot;response to interferon-beta&quot;, &quot;cellular response to interferon-beta&quot;, &quot;positive regulation of interferon-beta production&quot;, &quot;interferon-beta production&quot;, &quot;   
positive regulation of type I interferon production&quot;, &quot;regulation of type I interferon production&quot;, &quot;type I interferon production&quot;, &quot;cellular response to type I interferon&quot;, &quot;response to type I interferon&quot;, &quot;regulation of type I interferon-mediated signaling pathway&quot;, &quot;interferon-mediated signaling pathway&quot;, &quot;positive regulation of interferon-alpha production&quot;, &quot;nterferon-alpha production&quot;, &quot;regulation of interferon-alpha production&quot;, &quot;positive regulation of type I interferon-mediated signaling pathway&quot;, &quot;response to interferon-alpha&quot;, &quot;cellular response to interferon-alpha&quot;))



## geneID extraction
g2 &lt;- egoFb3$geneID
Str &lt;- unlist(strsplit(g2, &quot;/&quot;))
df &lt;- as.data.frame(Str)
colnames(df) &lt;- c(&quot;gene&quot;)

## signature Genes in count matrix
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)
if (nrow(signGenes) == 0) stop(&quot;No signature genes found.&quot;)

## sce object for signature genes
sceSub &lt;- sce[rownames(sce) %in% unique(signGenes$geneID), ]
if (nrow(sceSub) == 0) stop(&quot;No matching genes found in sce.&quot;)

## gene signature matrix
cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts))) / nrow(signGenes)
if (length(cntMat) != ncol(sceSub)) stop(&quot;Mismatch between cntMat and sceSub dimensions.&quot;)

sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign

sceSub$sign2[which(sceSub$sign &gt; 2)] &lt;- 2


sc &lt;- scale_colour_gradientn(colours = pal(100), limits = c(0, 2))

## UMAP all conditions for Interferon alpha pathway
plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis: ECM organisation
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-26-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()


## Donor Heart: ECM organisation
sceDHSub &lt;- sceDH[which(rownames(sceDH) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceDHSub@assays@data$logcounts)))/nrow(signGenes)
sceDHSub$sign &lt;- cntMat
sceDHSub$sign2 &lt;- sceDHSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceDHSub$sign2[which(sceDHSub$sign &gt; 1)] &lt;- 1

#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceDHSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-26-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="il-33-pathway" class="section level2">
<h2>IL-33 pathway</h2>
<pre class="r"><code>egoFb4 &lt;- dplyr::filter(egoFb@result, egoFb@result$Description %in% c(&quot;regulation of interleukin-33 production&quot;, &quot;  
interleukin-33 production&quot;))


## geneID extraction
g3 &lt;- egoFb4$geneID
Str &lt;- unlist(strsplit(g3, &quot;/&quot;))
df &lt;- as.data.frame(Str)
colnames(df) &lt;- c(&quot;gene&quot;)

## signature Genes in count matrix
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)
if (nrow(signGenes) == 0) stop(&quot;No signature genes found.&quot;)

## sce object for signature genes
sceSub &lt;- sce[rownames(sce) %in% unique(signGenes$geneID), ]
if (nrow(sceSub) == 0) stop(&quot;No matching genes found in sce.&quot;)

## gene signature matrix
cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts))) / nrow(signGenes)
if (length(cntMat) != ncol(sceSub)) stop(&quot;Mismatch between cntMat and sceSub dimensions.&quot;)

sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign

sceSub$sign2[which(sceSub$sign &gt; 2)] &lt;- 2


sc &lt;- scale_colour_gradientn(colours = pal(100), limits = c(0, 2))

## UMAP all conditions IL33
plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis: IL33
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-27-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()


## Donor Heart: IL33
sceDHSub &lt;- sceDH[which(rownames(sceDH) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceDHSub@assays@data$logcounts)))/nrow(signGenes)
sceDHSub$sign &lt;- cntMat
sceDHSub$sign2 &lt;- sceDHSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceDHSub$sign2[which(sceDHSub$sign &gt; 1)] &lt;- 1

#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceDHSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-27-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="interleukin-1-pathway" class="section level2">
<h2>Interleukin 1 pathway</h2>
<pre class="r"><code>egoFb5 &lt;- dplyr::filter(egoFb@result, egoFb@result$Description %in% c(&quot;response to interleukin-1&quot;, &quot;cellular response to interleukin-1&quot;, &quot;positive regulation of interleukin-1 alpha production&quot;, &quot;interleukin-1-mediated signaling pathway&quot;, &quot;interleukin-1 alpha production&quot;, &quot;regulation of interleukin-1 alpha production&quot;, &quot;positive regulation of interleukin-1 beta production&quot;, &quot;positive regulation of interleukin-1 production&quot;, &quot;    
interleukin-1 beta production&quot;, &quot;regulation of interleukin-1 beta production&quot;, &quot;negative regulation of interleukin-1 production&quot;, &quot;interleukin-1 production&quot;, &quot;regulation of interleukin-1 production&quot;, &quot;regulation of interleukin-1-mediated signaling pathway&quot;))


## geneID extraction
g4 &lt;- egoFb5$geneID
Str &lt;- unlist(strsplit(g4, &quot;/&quot;))
df &lt;- as.data.frame(Str)
colnames(df) &lt;- c(&quot;gene&quot;)

## signature Genes in count matrix
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)
if (nrow(signGenes) == 0) stop(&quot;No signature genes found.&quot;)

## sce object for signature genes
sceSub &lt;- sce[rownames(sce) %in% unique(signGenes$geneID), ]
if (nrow(sceSub) == 0) stop(&quot;No matching genes found in sce.&quot;)

## gene signature matrix
cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts))) / nrow(signGenes)
if (length(cntMat) != ncol(sceSub)) stop(&quot;Mismatch between cntMat and sceSub dimensions.&quot;)

sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign

sceSub$sign2[which(sceSub$sign &gt; 2)] &lt;- 2


sc &lt;- scale_colour_gradientn(colours = pal(100), limits = c(0, 2))

## UMAP all conditions Interleukin-1
plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis: Interleukin-1
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-28-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()


## Donor Heart: Interleukin-1
sceDHSub &lt;- sceDH[which(rownames(sceDH) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceDHSub@assays@data$logcounts)))/nrow(signGenes)
sceDHSub$sign &lt;- cntMat
sceDHSub$sign2 &lt;- sceDHSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceDHSub$sign2[which(sceDHSub$sign &gt; 1)] &lt;- 1

#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceDHSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-28-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="lymphocyte-differentiation" class="section level2">
<h2>Lymphocyte differentiation</h2>
<pre class="r"><code>egoFb6 &lt;- dplyr::filter(egoFb@result, egoFb@result$Description %in% c(&quot;lymphocyte differentiation&quot;, &quot;positive regulation of lymphocyte differentiation&quot;, &quot;regulation of lymphocyte differentiation&quot;, &quot;lymphocyte costimulation&quot;, &quot;lymphocyte migration&quot;, &quot;positive regulation of lymphocyte activation&quot;, &quot;lymphocyte activation involved in immune response&quot;, &quot;regulation of lymphocyte migration&quot;, &quot;lymphocyte chemotaxis&quot;, &quot;positive regulation of lymphocyte proliferation&quot;, &quot;regulation of lymphocyte proliferation&quot;, &quot;establishment of lymphocyte polarity&quot;, &quot; 
positive regulation of lymphocyte migration&quot;, &quot;positive regulation of lymphocyte chemotaxis&quot;, &quot;regulation of lymphocyte chemotaxis&quot;, &quot;regulation of lymphocyte mediated immunity&quot;))
                                                                    

## geneID extraction
g5 &lt;- egoFb6$geneID
Str &lt;- unlist(strsplit(g5, &quot;/&quot;))
df &lt;- as.data.frame(Str)
colnames(df) &lt;- c(&quot;gene&quot;)

## signature Genes in count matrix
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)
if (nrow(signGenes) == 0) stop(&quot;No signature genes found.&quot;)

## sce object for signature genes
sceSub &lt;- sce[rownames(sce) %in% unique(signGenes$geneID), ]
if (nrow(sceSub) == 0) stop(&quot;No matching genes found in sce.&quot;)

## gene signature matrix
cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts))) / nrow(signGenes)
if (length(cntMat) != ncol(sceSub)) stop(&quot;Mismatch between cntMat and sceSub dimensions.&quot;)

sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign

sceSub$sign2[which(sceSub$sign &gt; 2)] &lt;- 2


sc &lt;- scale_colour_gradientn(colours = pal(100), limits = c(0, 2))

## UMAP all conditions 
plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis: 
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-29-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()


## Donor Heart: 
sceDHSub &lt;- sceDH[which(rownames(sceDH) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceDHSub@assays@data$logcounts)))/nrow(signGenes)
sceDHSub$sign &lt;- cntMat
sceDHSub$sign2 &lt;- sceDHSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceDHSub$sign2[which(sceDHSub$sign &gt; 1)] &lt;- 1

#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceDHSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-29-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="bmp-signaling-pathway" class="section level2">
<h2>BMP signaling pathway</h2>
<pre class="r"><code>egoFb7 &lt;- dplyr::filter(egoFb@result, egoFb@result$Description %in% c(&quot;response to BMP&quot;, &quot;cellular response to BMP stimulus&quot;, &quot;regulation of BMP signaling pathway&quot;,     &quot;BMP signaling pathway&quot;, &quot;positive regulation of BMP signaling pathway&quot;))
                                                                    

## geneID extraction
g6 &lt;- egoFb7$geneID
Str &lt;- unlist(strsplit(g6, &quot;/&quot;))
df &lt;- as.data.frame(Str)
colnames(df) &lt;- c(&quot;gene&quot;)

## signature Genes in count matrix
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)
if (nrow(signGenes) == 0) stop(&quot;No signature genes found.&quot;)

## sce object for signature genes
sceSub &lt;- sce[rownames(sce) %in% unique(signGenes$geneID), ]
if (nrow(sceSub) == 0) stop(&quot;No matching genes found in sce.&quot;)

## gene signature matrix
cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts))) / nrow(signGenes)
if (length(cntMat) != ncol(sceSub)) stop(&quot;Mismatch between cntMat and sceSub dimensions.&quot;)

sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign

sceSub$sign2[which(sceSub$sign &gt; 2)] &lt;- 2


sc &lt;- scale_colour_gradientn(colours = pal(100), limits = c(0, 2))

## UMAP all conditions 
plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis: 
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-30-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()


## Donor Heart: 
sceDHSub &lt;- sceDH[which(rownames(sceDH) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceDHSub@assays@data$logcounts)))/nrow(signGenes)
sceDHSub$sign &lt;- cntMat
sceDHSub$sign2 &lt;- sceDHSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceDHSub$sign2[which(sceDHSub$sign &gt; 1)] &lt;- 1

#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceDHSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-30-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="innate-immune-response" class="section level2">
<h2>Innate immune response</h2>
<pre class="r"><code>egoTRM1 &lt;- dplyr::filter(egoTRM@result, egoTRM@result$Description %in% c(&quot;innate immune response activating cell surface receptor signaling pathway&quot;, &quot;activation of innate immune response&quot;, &quot;innate immune response-activating signaling pathway&quot;, &quot;innate immune response in mucosa&quot;))


## geneID extraction
g8 &lt;- egoTRM1$geneID
Str &lt;- unlist(strsplit(g8, &quot;/&quot;))
df &lt;- as.data.frame(Str)
colnames(df) &lt;- c(&quot;gene&quot;)

## signature Genes in count matrix
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)
if (nrow(signGenes) == 0) stop(&quot;No signature genes found.&quot;)

## sce object for signature genes
sceSub &lt;- sce[rownames(sce) %in% unique(signGenes$geneID), ]
if (nrow(sceSub) == 0) stop(&quot;No matching genes found in sce.&quot;)

## gene signature matrix
cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts))) / nrow(signGenes)
if (length(cntMat) != ncol(sceSub)) stop(&quot;Mismatch between cntMat and sceSub dimensions.&quot;)

sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign

sceSub$sign2[which(sceSub$sign &gt; 2)] &lt;- 2


sc &lt;- scale_colour_gradientn(colours = pal(100), limits = c(0, 2))

## UMAP all conditions
plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis 
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-31-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()

## Donor Heart: 
sceDHSub &lt;- sceDH[which(rownames(sceDH) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceDHSub@assays@data$logcounts)))/nrow(signGenes)
sceDHSub$sign &lt;- cntMat
sceDHSub$sign2 &lt;- sceDHSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceDHSub$sign2[which(sceDHSub$sign &gt; 1)] &lt;- 1

#pdf(&quot;umap_.pdf&quot;, width = 8, height = 6) 
plotUMAP(sceDHSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Sarcoidosis_allClusters_Validation_Nature.Rmd/unnamed-chunk-31-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Sun Oct  5 17:57:10 2025&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_2.0.4              future_1.67.0               workflowr_1.7.2            
 [4] pathview_1.46.0             wordcloud_2.6               textshaping_1.0.3          
 [7] EnhancedVolcano_1.24.0      ggrepel_0.9.6               RColorBrewer_1.1-3         
[10] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
[13] ggupset_0.4.1               gridExtra_2.3               DOSE_4.0.1                 
[16] enrichplot_1.26.6           msigdbr_25.1.1              org.Hs.eg.db_3.20.0        
[19] AnnotationDbi_1.68.0        clusterProfiler_4.14.6      multtest_2.62.0            
[22] metap_1.12                  scater_1.34.1               scuttle_1.16.0             
[25] destiny_3.20.0              circlize_0.4.16             muscat_1.20.0              
[28] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[31] forcats_1.0.1               stringr_1.5.2               purrr_1.1.0                
[34] readr_2.1.5                 tidyr_1.3.1                 tibble_3.3.0               
[37] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.28.1
[40] SummarizedExperiment_1.36.0 Biobase_2.66.0              GenomicRanges_1.58.0       
[43] GenomeInfoDb_1.42.3         IRanges_2.40.1              S4Vectors_0.44.0           
[46] BiocGenerics_0.52.0         MatrixGenerics_1.18.1       matrixStats_1.5.0          
[49] pheatmap_1.0.13             ggpubr_0.6.1                ggplot2_4.0.0              
[52] Seurat_5.3.0                SeuratObject_5.2.0          sp_2.2-0                   

loaded via a namespace (and not attached):
  [1] graph_1.84.1             igraph_2.1.4             ica_1.0-3               
  [4] plotly_4.11.0            Formula_1.2-5            zlibbioc_1.52.0         
  [7] tidyselect_1.2.1         bit_4.6.0                doParallel_1.0.17       
 [10] clue_0.3-66              lattice_0.22-7           rjson_0.2.23            
 [13] blob_1.2.4               S4Arrays_1.6.0           pbkrtest_0.5.5          
 [16] parallel_4.4.2           png_0.1-8                plotrix_3.8-4           
 [19] cli_3.6.5                ggplotify_0.1.3          goftest_1.2-3           
 [22] VIM_6.2.6                variancePartition_1.36.3 BiocNeighbors_2.0.1     
 [25] uwot_0.2.3               curl_7.0.0               mime_0.13               
 [28] evaluate_1.0.5           tidytree_0.4.6           ComplexHeatmap_2.22.0   
 [31] stringi_1.8.7            backports_1.5.0          XML_3.99-0.19           
 [34] lmerTest_3.1-3           qqconf_1.3.2             httpuv_1.6.16           
 [37] rappdirs_0.3.3           splines_4.4.2            sctransform_0.4.2       
 [40] ggbeeswarm_0.7.2         DBI_1.2.3                jquerylib_0.1.4         
 [43] smoother_1.3             withr_3.0.2              git2r_0.36.2            
 [46] corpcor_1.6.10           systemfonts_1.2.3        reformulas_0.4.1        
 [49] class_7.3-23             rprojroot_2.1.1          lmtest_0.9-40           
 [52] formatR_1.14             htmlwidgets_1.6.4        fs_1.6.6                
 [55] labeling_0.4.3           fANCOVA_0.6-1            SparseArray_1.6.2       
 [58] DESeq2_1.46.0            ranger_0.17.0            DEoptimR_1.1-4          
 [61] reticulate_1.43.0        hexbin_1.28.5            zoo_1.8-14              
 [64] XVector_0.46.0           knitr_1.50               ggplot.multistats_1.0.1 
 [67] UCSC.utils_1.2.0         RhpcBLASctl_0.23-42      timechange_0.3.0        
 [70] foreach_1.5.2            patchwork_1.3.2          caTools_1.18.3          
 [73] data.table_1.17.8        ggtree_3.14.0            R.oo_1.27.1             
 [76] RSpectra_0.16-2          irlba_2.3.5.1            ggrastr_1.0.2           
 [79] fastDummies_1.7.5        gridGraphics_0.5-1       lazyeval_0.2.2          
 [82] yaml_2.3.10              survival_3.8-3           scattermore_1.2         
 [85] crayon_1.5.3             RcppAnnoy_0.0.22         Rgraphviz_2.50.0        
 [88] progressr_0.16.0         later_1.4.4              ggridges_0.5.7          
 [91] codetools_0.2-20         GlobalOptions_0.1.2      aod_1.3.3               
 [94] KEGGREST_1.46.0          Rtsne_0.17               shape_1.4.6.1           
 [97] limma_3.62.2             estimability_1.5.1       pkgconfig_2.0.3         
[100] KEGGgraph_1.66.0         TMB_1.9.17               spatstat.univar_3.1-4   
[103] mathjaxr_1.8-0           getPass_0.2-4            EnvStats_3.1.0          
[106] aplot_0.2.9              scatterplot3d_0.3-44     spatstat.sparse_3.1-0   
[109] ape_5.8-1                xtable_1.8-4             car_3.1-3               
[112] plyr_1.8.9               httr_1.4.7               rbibutils_2.3           
[115] tools_4.4.2              globals_0.18.0           beeswarm_0.4.0          
[118] broom_1.0.10             nlme_3.1-168             lambda.r_1.2.4          
[121] assertthat_0.2.1         lme4_1.1-37              digest_0.6.37           
[124] numDeriv_2016.8-1.1      Matrix_1.7-4             farver_2.1.2            
[127] tzdb_0.5.0               remaCor_0.0.20           reshape2_1.4.4          
[130] yulab.utils_0.2.1        glue_1.8.0               cachem_1.1.0            
[133] polyclip_1.10-7          generics_0.1.4           Biostrings_2.74.1       
[136] mvtnorm_1.3-3            presto_1.0.0             parallelly_1.45.1       
[139] mnormt_2.1.1             statmod_1.5.0            RcppHNSW_0.6.0          
[142] ScaledMatrix_1.14.0      carData_3.0-5            minqa_1.2.8             
[145] pbapply_1.7-4            spam_2.11-1              gson_0.1.0              
[148] gtools_3.9.5             ggsignif_0.6.4           RcppEigen_0.3.4.0.2     
[151] shiny_1.11.1             GenomeInfoDbData_1.2.13  glmmTMB_1.1.12          
[154] R.utils_2.13.0           RCurl_1.98-1.17          memoise_2.0.1           
[157] rmarkdown_2.29           scales_1.4.0             R.methodsS3_1.8.2       
[160] RANN_2.6.2               spatstat.data_3.1-8      rstudioapi_0.17.1       
[163] whisker_0.4.1            cluster_2.1.8.1          mutoss_0.1-13           
[166] spatstat.utils_3.2-0     hms_1.1.3                fitdistrplus_1.2-4      
[169] cowplot_1.2.0            colorspace_2.1-2         rlang_1.1.6             
[172] xts_0.14.1               dotCall64_1.2            ggtangle_0.0.7          
[175] laeken_0.5.3             mgcv_1.9-3               xfun_0.53               
[178] coda_0.19-4.1            e1071_1.7-16             TH.data_1.1-4           
[181] iterators_1.0.14         emmeans_1.11.2-8         abind_1.4-8             
[184] GOSemSim_2.32.0          treeio_1.30.0            ps_1.9.1                
[187] futile.options_1.0.1     bitops_1.0-9             Rdpack_2.6.4            
[190] promises_1.3.3           RSQLite_2.4.3            qvalue_2.38.0           
[193] sandwich_3.1-1           fgsea_1.32.4             DelayedArray_0.32.0     
[196] proxy_0.4-27             GO.db_3.20.0             compiler_4.4.2          
[199] prettyunits_1.2.0        boot_1.3-32              beachmat_2.22.0         
[202] listenv_0.9.1            Rcpp_1.1.0               edgeR_4.4.2             
[205] BiocSingular_1.22.0      tensor_1.5.1             MASS_7.3-65             
[208] progress_1.2.3           BiocParallel_1.40.2      babelgene_22.9          
[211] spatstat.random_3.4-2    R6_2.6.1                 fastmap_1.2.0           
[214] multcomp_1.4-28          fastmatch_1.1-6          rstatix_0.7.2           
[217] vipor_0.4.7              TTR_0.24.4               ROCR_1.0-11             
[220] TFisher_0.2.0            rsvd_1.0.5               vcd_1.4-13              
[223] nnet_7.3-20              gtable_0.3.6             KernSmooth_2.23-26      
[226] miniUI_0.1.2             deldir_2.0-4             htmltools_0.5.8.1       
[229] ggthemes_5.1.0           bit64_4.6.0-1            spatstat.explore_3.5-3  
[232] lifecycle_1.0.4          blme_1.0-6               S7_0.2.0                
[235] processx_3.8.6           callr_3.7.6              nloptr_2.2.1            
[238] sass_0.4.10              vctrs_0.6.5              robustbase_0.99-6       
[241] spatstat.geom_3.6-0      sn_2.1.1                 ggfun_0.2.0             
[244] future.apply_1.20.0      bslib_0.9.0              pillar_1.11.1           
[247] gplots_3.2.0             pcaMethods_1.98.0        locfit_1.5-9.12         
[250] jsonlite_2.0.0           GetoptLong_1.0.5        </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_2.0.4              future_1.67.0               workflowr_1.7.2            
 [4] pathview_1.46.0             wordcloud_2.6               textshaping_1.0.3          
 [7] EnhancedVolcano_1.24.0      ggrepel_0.9.6               RColorBrewer_1.1-3         
[10] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
[13] ggupset_0.4.1               gridExtra_2.3               DOSE_4.0.1                 
[16] enrichplot_1.26.6           msigdbr_25.1.1              org.Hs.eg.db_3.20.0        
[19] AnnotationDbi_1.68.0        clusterProfiler_4.14.6      multtest_2.62.0            
[22] metap_1.12                  scater_1.34.1               scuttle_1.16.0             
[25] destiny_3.20.0              circlize_0.4.16             muscat_1.20.0              
[28] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[31] forcats_1.0.1               stringr_1.5.2               purrr_1.1.0                
[34] readr_2.1.5                 tidyr_1.3.1                 tibble_3.3.0               
[37] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.28.1
[40] SummarizedExperiment_1.36.0 Biobase_2.66.0              GenomicRanges_1.58.0       
[43] GenomeInfoDb_1.42.3         IRanges_2.40.1              S4Vectors_0.44.0           
[46] BiocGenerics_0.52.0         MatrixGenerics_1.18.1       matrixStats_1.5.0          
[49] pheatmap_1.0.13             ggpubr_0.6.1                ggplot2_4.0.0              
[52] Seurat_5.3.0                SeuratObject_5.2.0          sp_2.2-0                   

loaded via a namespace (and not attached):
  [1] graph_1.84.1             igraph_2.1.4             ica_1.0-3               
  [4] plotly_4.11.0            Formula_1.2-5            zlibbioc_1.52.0         
  [7] tidyselect_1.2.1         bit_4.6.0                doParallel_1.0.17       
 [10] clue_0.3-66              lattice_0.22-7           rjson_0.2.23            
 [13] blob_1.2.4               S4Arrays_1.6.0           pbkrtest_0.5.5          
 [16] parallel_4.4.2           png_0.1-8                plotrix_3.8-4           
 [19] cli_3.6.5                ggplotify_0.1.3          goftest_1.2-3           
 [22] VIM_6.2.6                variancePartition_1.36.3 BiocNeighbors_2.0.1     
 [25] uwot_0.2.3               curl_7.0.0               mime_0.13               
 [28] evaluate_1.0.5           tidytree_0.4.6           ComplexHeatmap_2.22.0   
 [31] stringi_1.8.7            backports_1.5.0          XML_3.99-0.19           
 [34] lmerTest_3.1-3           qqconf_1.3.2             httpuv_1.6.16           
 [37] rappdirs_0.3.3           splines_4.4.2            sctransform_0.4.2       
 [40] ggbeeswarm_0.7.2         DBI_1.2.3                jquerylib_0.1.4         
 [43] smoother_1.3             withr_3.0.2              git2r_0.36.2            
 [46] corpcor_1.6.10           systemfonts_1.2.3        reformulas_0.4.1        
 [49] class_7.3-23             rprojroot_2.1.1          lmtest_0.9-40           
 [52] formatR_1.14             htmlwidgets_1.6.4        fs_1.6.6                
 [55] labeling_0.4.3           fANCOVA_0.6-1            SparseArray_1.6.2       
 [58] DESeq2_1.46.0            ranger_0.17.0            DEoptimR_1.1-4          
 [61] reticulate_1.43.0        hexbin_1.28.5            zoo_1.8-14              
 [64] XVector_0.46.0           knitr_1.50               ggplot.multistats_1.0.1 
 [67] UCSC.utils_1.2.0         RhpcBLASctl_0.23-42      timechange_0.3.0        
 [70] foreach_1.5.2            patchwork_1.3.2          caTools_1.18.3          
 [73] data.table_1.17.8        ggtree_3.14.0            R.oo_1.27.1             
 [76] RSpectra_0.16-2          irlba_2.3.5.1            ggrastr_1.0.2           
 [79] fastDummies_1.7.5        gridGraphics_0.5-1       lazyeval_0.2.2          
 [82] yaml_2.3.10              survival_3.8-3           scattermore_1.2         
 [85] crayon_1.5.3             RcppAnnoy_0.0.22         Rgraphviz_2.50.0        
 [88] progressr_0.16.0         later_1.4.4              ggridges_0.5.7          
 [91] codetools_0.2-20         GlobalOptions_0.1.2      aod_1.3.3               
 [94] KEGGREST_1.46.0          Rtsne_0.17               shape_1.4.6.1           
 [97] limma_3.62.2             estimability_1.5.1       pkgconfig_2.0.3         
[100] KEGGgraph_1.66.0         TMB_1.9.17               spatstat.univar_3.1-4   
[103] mathjaxr_1.8-0           getPass_0.2-4            EnvStats_3.1.0          
[106] aplot_0.2.9              scatterplot3d_0.3-44     spatstat.sparse_3.1-0   
[109] ape_5.8-1                xtable_1.8-4             car_3.1-3               
[112] plyr_1.8.9               httr_1.4.7               rbibutils_2.3           
[115] tools_4.4.2              globals_0.18.0           beeswarm_0.4.0          
[118] broom_1.0.10             nlme_3.1-168             lambda.r_1.2.4          
[121] assertthat_0.2.1         lme4_1.1-37              digest_0.6.37           
[124] numDeriv_2016.8-1.1      Matrix_1.7-4             farver_2.1.2            
[127] tzdb_0.5.0               remaCor_0.0.20           reshape2_1.4.4          
[130] yulab.utils_0.2.1        glue_1.8.0               cachem_1.1.0            
[133] polyclip_1.10-7          generics_0.1.4           Biostrings_2.74.1       
[136] mvtnorm_1.3-3            presto_1.0.0             parallelly_1.45.1       
[139] mnormt_2.1.1             statmod_1.5.0            RcppHNSW_0.6.0          
[142] ScaledMatrix_1.14.0      carData_3.0-5            minqa_1.2.8             
[145] pbapply_1.7-4            spam_2.11-1              gson_0.1.0              
[148] gtools_3.9.5             ggsignif_0.6.4           RcppEigen_0.3.4.0.2     
[151] shiny_1.11.1             GenomeInfoDbData_1.2.13  glmmTMB_1.1.12          
[154] R.utils_2.13.0           RCurl_1.98-1.17          memoise_2.0.1           
[157] rmarkdown_2.29           scales_1.4.0             R.methodsS3_1.8.2       
[160] RANN_2.6.2               spatstat.data_3.1-8      rstudioapi_0.17.1       
[163] whisker_0.4.1            cluster_2.1.8.1          mutoss_0.1-13           
[166] spatstat.utils_3.2-0     hms_1.1.3                fitdistrplus_1.2-4      
[169] cowplot_1.2.0            colorspace_2.1-2         rlang_1.1.6             
[172] xts_0.14.1               dotCall64_1.2            ggtangle_0.0.7          
[175] laeken_0.5.3             mgcv_1.9-3               xfun_0.53               
[178] coda_0.19-4.1            e1071_1.7-16             TH.data_1.1-4           
[181] iterators_1.0.14         emmeans_1.11.2-8         abind_1.4-8             
[184] GOSemSim_2.32.0          treeio_1.30.0            ps_1.9.1                
[187] futile.options_1.0.1     bitops_1.0-9             Rdpack_2.6.4            
[190] promises_1.3.3           RSQLite_2.4.3            qvalue_2.38.0           
[193] sandwich_3.1-1           fgsea_1.32.4             DelayedArray_0.32.0     
[196] proxy_0.4-27             GO.db_3.20.0             compiler_4.4.2          
[199] prettyunits_1.2.0        boot_1.3-32              beachmat_2.22.0         
[202] listenv_0.9.1            Rcpp_1.1.0               edgeR_4.4.2             
[205] BiocSingular_1.22.0      tensor_1.5.1             MASS_7.3-65             
[208] progress_1.2.3           BiocParallel_1.40.2      babelgene_22.9          
[211] spatstat.random_3.4-2    R6_2.6.1                 fastmap_1.2.0           
[214] multcomp_1.4-28          fastmatch_1.1-6          rstatix_0.7.2           
[217] vipor_0.4.7              TTR_0.24.4               ROCR_1.0-11             
[220] TFisher_0.2.0            rsvd_1.0.5               vcd_1.4-13              
[223] nnet_7.3-20              gtable_0.3.6             KernSmooth_2.23-26      
[226] miniUI_0.1.2             deldir_2.0-4             htmltools_0.5.8.1       
[229] ggthemes_5.1.0           bit64_4.6.0-1            spatstat.explore_3.5-3  
[232] lifecycle_1.0.4          blme_1.0-6               S7_0.2.0                
[235] processx_3.8.6           callr_3.7.6              nloptr_2.2.1            
[238] sass_0.4.10              vctrs_0.6.5              robustbase_0.99-6       
[241] spatstat.geom_3.6-0      sn_2.1.1                 ggfun_0.2.0             
[244] future.apply_1.20.0      bslib_0.9.0              pillar_1.11.1           
[247] gplots_3.2.0             pcaMethods_1.98.0        locfit_1.5-9.12         
[250] jsonlite_2.0.0           GetoptLong_1.0.5        </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
